src/components/Actions.vue

```diff
@@ -53,14 +53,31 @@ export default {
     error() {
       return this.$store.state.error
     },
+    userJourney() {
+      return this.$store.getters.getUserJourney
+    },
+    previousStep() {
+      return this.$state.previous(
+        this.$route,
+        this.$store.state.situation,
+        this.$router
+      )
+    },
   },
   methods: {
     localOnSubmit: function (event) {
       event.preventDefault()
       this.onSubmit()
     },
     goBack: function () {
-      window && window.history.back()
+      if (
+        this.userJourney.history[this.userJourney.history.length - 2] ===
+        this.previousStep.path
+      ) {
+        window && window.history.back()
+      } else {
+        this.$pop()
+      }
```

Comment:
I don't get this part ! 
if the previous userJourney is the same as the previousStep path (isn't it always the case ?) we go back
if not we are going back by an vuex event ?

Reply:
On réalise cette condition pour éviter de modifier l'historique quand l'utilisateur retourne sur la route précédente dans le cas où il suit le parcours classique.
Dans le cas où l'utilisateur utilise le sommaire pour se déplacer l'historique n'est plus équivalent à la step précédente, dans ce cas on utilise pop.

Voici un exemple :
CAS 1 :
l'utilisateur est sur la page logement.
Il fait "précédent" et se retrouve sur la page précédente.
Dans ce cas on utilise`window.history.back()`.

CAS 2 :
L'utilisateur est sur la page logement.
Il clique sur "Mon foyer" du sommaire et clique sur le bouton précédent.
Dans ce cas nous utilisons la fonction `$pop` pour revenir à la page `_boursier_derniere_annee_etudes`.
L'utilisation de `window.history.back()` est impossible, elle nous ferait revenir à la page précédente de l'historique  : logement.